### PR TITLE
Github actions

### PR DESCRIPTION
**Pull request by Scot3004**
Run automated checks in github actions, all check should pass in github

### CODE REVIEW

File: pyproject.toml
Comment (bandophahita) [inline pyproject.toml]: My fault for not fixing the original comment in screenpy and screenpy_selenium.  This really SHOULD say `poetry install --extras dev_all`

**Comment by Scot3004** [comment]
I don't know why the checks are not being displayed here but, the jobs are running
https://github.com/Scot3004/screenpy_adapter_allure/actions

All the actions are passing now
image

**Comment by bandophahita** [comment]
> I don't know why the checks are not being displayed here...

They are but the UX is a little confusing at first.  The actions for Lint, Tests, and Poetry are under each run.  The run names are auto generated based on what branch they occur in.   
image

 And then if you click on a run, you'll see each environment job.  
image

Click on one of the environment jobs, you'll see the actual break-down of the job.
image

**Comment by Scot3004** [comment]
Correct, but when a PR is created it should display the result of the actions here 
image

**Comment by Scot3004** [comment]
Fun fact, The action Test passed at the first try, the other, required some tune up and learn new stuff like poetry and the pyproject.toml 😂

File: pyproject.toml
Comment (perrygoy) [inline pyproject.toml]: Hey @Scot3004, wanna add your name to this list? 😉

File: pyproject.toml
Comment (perrygoy) [inline pyproject.toml]: I think we can get rid of these two lines; they're dependencies for ScreenPy but not for this adapater.

File: pyproject.toml
Comment (perrygoy) [inline pyproject.toml:2]: Oh, that reminds me—can you add py311 to the `tox.ini` env list in this PR? 🙏

**Review by perrygoy — COMMENTED** [review]
I have a few minor things to consider, any changes would dismiss my approval—so i'm just commenting for now!

Could the tests not be showing up for this PR because they aren't merged into `trunk` yet? We'll have to see if the next PR after this one is missing them, too.

**Comment by bandophahita** [comment]
> Correct, but when a PR is created it should display the result of the actions here

Ahh... I think that might be due to the fact we haven't added the actions to the trunk yet. Seem to recall something like that happening when I added them to my branch (for screenpy)  before we merged the first time.  

Edit: Confirmed.  The first time the github actions were added to the branch the PR didn't show the checks.  I've never looked into a way around that.

File: pyproject.toml
Comment (Scot3004) [reply pyproject.toml]: Yes 😄

**Comment by Scot3004** [comment]
Added a new commit, note I added importlib.metadata to have a single source of truth, because the values are the same than the ones on pyproject.toml

**Review by perrygoy — APPROVED** [review]
Thanks for the quick fixes, @Scot3004—looks good to me!

**Pull request merged**
